daemon/graphdriver/driver.go

```diff
@@ -85,6 +86,26 @@ type Driver interface {
 	DiffSize(id, parent string) (size int64, err error)
 }
 
+// CustomImageAccessor is an interface to allow callers to query the object for
+// any implementation specific image information that might need to be inserted into
+// the graph.
+type CustomImageAccessor interface {
+	// GetCustomImages returns a json string with metadata describing the images
+	// known to this implementation. The data is a json encoded CustomImageInfoList.
+	GetCustomImages() (string, error)
+}
+
+type CustomImageInfo struct {
+	Name        string
+	Version     string
+	Path        string
+	Size        int64
+	CreatedTime time.Time
+}
+type CustomImageInfoList struct {
+	Images []CustomImageInfo
+}
+
```

Comment:
I wasn't quite happy with adding these here... I know we originally hoped that the code for adding the images to the graph would live in the graphdriver only, but I couldn't figure out how to make that work (see comment on that file).  Since the CustomImageAccessor interface was public, I thought it made more sense to also make the structs describing the format of the returned JSON public, but that leans to far toward making this general infrastructure as opposed to something driver specific.  I'd hope we can fix this along with the other method in graph.go.